src/sonic-yang-models/yang-models/sonic-snmp.yang

```diff
@@ -156,5 +162,35 @@ module sonic-snmp {
         }
       }
     }
+    container SNMP_AGENT_ADDRESS_CONFIG {
+      list SNMP_AGENT_ADDRESS_LIST {
+        key "agent_ip port vrf_name";
+        unique "agent_ip port";
+        description "List of SNMP agent listening IP Addresses and ports.";
+
+        leaf agent_ip {
+          type inet:ip-address;
+          description "SNMP agent listening IP";
+        }
+        leaf port {
+          type inet:port-number;
+          description "SNMP agent listening port number";
+        }
+        leaf vrf_name {
+          type union {
+            type string {
+              pattern '';
```

Comment:
>pattern [](http://example.com/codeflow?start=14&length=7)

@ganglyu @wen587  I find empty string is weird. Is it going to impact gnmi path, impact json representation of ConfigDB?

Reply:
The reason for empty string is because config CLI adds key in CONFIG_TABLE with empty string for vrf if it is not specified in the CLI argument.
https://github.com/sonic-net/sonic-utilities/blob/49fc3896a0ef907ff337467a3e04d2501ef850fd/config/main.py#L3084

Reply:
GNMI path can support it, need GCU to confirm.

Reply:
GCU can support it.